### ISSUE TITLE
Issue with data files when there is not data for all energy bins

### ISSUE THREAD

**Issue by hayesla** [samaloney/stixpy#40]
When the data files do not have data for all 32 energy bins then the `ScienceData.get_data()` has issues with array shapes as the energies are not masked in this function 

For example:

```python
>>> stix_file1 = 'solo_L1_stix-sci-xray-cpd_20220215T214529-20220215T222336_V01_2202154670-60563.fits'
>>> stix_data1 = ScienceData.from_fits(stix_file1)
>>> stix_data1.get_data()

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/var/folders/wm/gysdzbp10qv1kcm4zxb5q95xkjzv5q/T/ipykernel_26628/1532764551.py in <module>
----> 1 stix_data1.get_data()

~/dev_things/stixpy/stixpy/science.py in get_data(self, time_indices, energy_indices, detector_indices, pixel_indices, sum_all_times)
    539             t_norm = t_norm.reshape(-1, 1, 1, 1)
    540 
--> 541         counts_err = np.sqrt(counts*u.ct + counts_var)/(e_norm * t_norm)
    542         counts = counts / (e_norm * t_norm)
    543 

~/opt/miniconda3/envs/stix_dev/lib/python3.9/site-packages/astropy/units/quantity.py in __truediv__(self, other)
   1149                 return NotImplemented
   1150 
-> 1151         return super().__truediv__(other)
   1152 
   1153     def __itruediv__(self, other):

~/opt/miniconda3/envs/stix_dev/lib/python3.9/site-packages/astropy/units/quantity.py in __array_ufunc__(self, function, method, *inputs, **kwargs)
    612 
    613         # Call our superclass's __array_ufunc__
--> 614         result = super().__array_ufunc__(function, method, *arrays, **kwargs)
    615         # If unit is None, a plain array is expected (e.g., comparisons), which
    616         # means we're done.

ValueError: operands could not be broadcast together with shapes (137,32,12,17) (137,1,1,32) 
```

The reason being that if we look at the `data.counts` the shape is `(137,32,12,17)` whereas the energies are of size 32 rather than 17. The energy bin mask that is in control needs to be used here.

**Comment by samaloney** [comment]
I think this is fixed by change the `Spectrogram.__init__`  method from this 
https://github.com/samaloney/stixpy/blob/e41582be8b6d4838fa25ef9cb81771cc955d8fc4/stixpy/science.py#L865-L884
to
`self.dE = (energies['e_high'] - energies['e_low'])[self.energy_masks.masks[0] == 1]`

**Comment by samaloney** [comment]
Need to check with #50

**Comment by hayesla** [comment]
fixed now by #50
